src/graph/edge.cc

```diff
@@ -281,26 +281,33 @@ namespace hpp {
 			ConfigurationIn_t q2)
 	const
       {
-        ConstraintSetPtr_t constraints = pathConstraint ();
-        constraints->configProjector ()->rightHandSideFromConfig(q1);
-        if (constraints->isSatisfied (q1)) {
-          if (constraints->isSatisfied (q2)) {
-            path = (*steeringMethod_->get()) (q1, q2);
-            return (bool)path;
-          }
-          hppDout(info, "q2 does not satisfy the constraints");
-        }
 	core::SteeringMethodPtr_t sm (steeringMethod_->get());
 	if (!sm) {
 	  buildPathConstraint ();
 	}
+	sm = (steeringMethod_->get());
 	if (!sm) {
 	  std::ostringstream oss;
 	  oss << "No steering method set in edge " << name () << ".";
 	  throw std::runtime_error (oss.str ().c_str ());
 	}
-	path = (*sm) (q1, q2);
-        return path;
+        ConstraintSetPtr_t constraints = pathConstraint ();
```

Comment:
Though not very important, the bug you had with the steering method being null should not happen after a call to `pathConstraint()` since https://github.com/humanoid-path-planner/hpp-manipulation/commit/b40617253ce3232a7d9f91435baf91f7a7c52fd8#diff-70f90d9d3f3dcacc7740c890a847fbdb
. `pathConstraints` calls `buildPathConstraints` when it is needed. So the lines
```cpp
core::SteeringMethodPtr_t sm (steeringMethod_->get());
 if (!sm) {
   buildPathConstraint ();
 }
 sm = (steeringMethod_->get());
 if (!sm) {
   std::ostringstream oss;
   oss << "No steering method set in edge " << name () << ".";
   throw std::runtime_error (oss.str ().c_str ());
 }
```
should be superfluous.